x-pack/plugins/alerting/server/saved_objects/partially_update_rule.ts

```diff
@@ -67,3 +70,33 @@ export async function partiallyUpdateRule(
     throw err;
   }
 }
+
+// direct, partial update to a rule saved object via ElasticsearchClient
+export async function partiallyUpdateRuleWithEs(
+  esClient: ElasticsearchClient,
+  id: string,
+  attributes: PartiallyUpdateableRuleAttributes,
+  options: PartiallyUpdateRuleSavedObjectOptions = {}
+): Promise<void> {
+  // ensure we only have the valid attributes that are not encrypted and are excluded from AAD
+  const attributeUpdates = omit(attributes, [
```

Comment:
This seems slighty dangerous, in case we forget to include one of the attribues in the lists used below.

Feels like it might be better to have an explicit list of attributes that we ALLOW to be partially updated.

Reply:
++

Reply:
I agree as well. Filtering by an explicit safe list of just the attributes that we expect to modify in this way, and also omitting any encrypted/AAD attributes (just in case), seems most prudent. Even if it requires a little bit more maintenance in the long run, this has potential to go quite poorly, so we should do what we can to prevent that possibility.

Reply:
@ymao1 Can we wait to merge this until Oleg has a chance to take a look?

Reply:
@jeramysoucy Yes absolutely! I'll make the changes for the AAD attributes and will wait to merge

Reply:
Added allowlist in this commit: https://github.com/elastic/kibana/pull/193341/commits/468307830e7ff7d368adf972162708273bebcc65

Reply:
Perfect!

Reply:
Discussed with @azasypkin and he is 👍 with an additional comment about why we're bypassing audit logging (added in https://github.com/elastic/kibana/pull/193341/commits/409a2faa3cbe53161729292eb967358586ac9682) and a followup issue for the Core team to investigate how they might expose a more performance SO client (https://github.com/elastic/kibana/issues/194435)